### DISCUSSION TITLE
Error: context deadline exceeded (WINDOWS)

### DISCUSSION THREAD

**Discussion by CrxsTagz** [netbirdio/netbird#300] [Q&A / Support] [answered]

**Comment by CrxsTagz** [comment]
I SOLVED IT

**Reply by brandonros** [reply]
HOW

**Reply by mlsmaycon** [reply]
Our apologies for missing this discussion. 

here is my answer for documenting and future issues:

on client we get "Error: context deadline exceeded" in 2 occasions:

1 - Daemon is not running, so you should run the `netbird service install` and `netbird service start`
2 - connectivity issues with management and signal, in these case, if you are running self-hosted, you should make sure that you are passing the correct `--management-url http(s)://ADDDRESS:33073` and that both services have their firewall allowing tcp ports 33073 and 10000

**Reply by rchavers** [reply]
> I SOLVED IT

How did you solve it?

I am also having this issue on my self-hosted netbird.  netbird v0.8.6, ubuntu 22.04, installed on digital ocean droplet.  I'm seeing this problem on both linux and windows 10 clients, I can't figure out what I'm doing wrong.  My LE certs are working, I can log in to my netbird, my docker logs show no matching errors, and mismaycon's fix doesn't seem to work for me.

I'm stumped.

**Reply by braginini** [reply]
@rchavers let's debug it together on Slack. Usually, there is some minor issue with the configuration.
[NetBird Slack Channel](https://join.slack.com/t/netbirdio/shared_invite/zt-vrahf41g-ik1v7fV8du6t0RwxSrJ96A)

**Comment by rchavers** [comment]
Misha,

Thanks for the offer, I've been away on a business trip.  However, I
noticed an update to v0.8.7 and that update fixed my issue.  I can now
happily connect peers to my self-hosted netbird server.  Thanks!

**Reply by braginini** [reply]
Super @rchavers 
Thank you for the feedback!

**Comment by braginini** [comment, marked as answer]
The issue was related to Management not being able to server gRPC requests from agents due to the SSL cert loading issue.
https://github.com/netbirdio/netbird/pull/423

An update to v0.8.7 solves the problem.
